added suport for proxy per request

sean98 opened the pull request:
Now it is possible to select a different proxy for each request
by doing:
```java
Stripe.apiKey = "sk_test_...";

RequestOptions requestOptions = RequestOptions.builder()
        .setConnectionProxy(<java.net.Proxy>)
        .setProxyCredential(<java.net.PasswordAuthentication>)
        .build();

Balance.retrieve(requestOptions);
```

richardm-stripe commented:
Hi @sean98! Thanks for the contribution -- I think this makes sense and will be taking a closer look at this today. Would you be willing to modify `src/test/java/com/stripe/net/RequestOptionsTest.java` to exercise the new setters?

richardm-stripe commented:
Also I would be curious to learn what your use case for the per-request proxy is.

sean98 commented:
Hey! 
I updated ```src/test/java/com/stripe/net/RequestOptionsTest.java``` and added a few tests, I tried to make them consistent with the current ones.

The reason I need this feature is that I'm working with a proxy service that has data limitations per proxy, that why I'm using multiple proxies in the same time to balance the data and wanted an easy way to change the current proxy.

richardm-stripe commented:
Thank you @sean98 for the contribution! It's been merged and we released in [v19.37.0](https://github.com/stripe/stripe-java/releases/tag/v19.37.0)
